src/intrinsics/ecma262/Object.js

```diff
@@ -140,7 +140,16 @@ export default function(realm: Realm): NativeFunctionValue {
     }
 
     // 5. Return to.
-    if (to_must_be_partial) to.makePartial();
+    if (to_must_be_partial) {
+      const to_was_simple = to.isSimpleObject();
```

Comment:
simple and partial should be orthogonal, so don't make the simplicity of to subject to it being partial.